crates/extensions/c8y_firmware_manager/src/config.rs

```diff
@@ -19,11 +20,11 @@ const FIRMWARE_UPDATE_RESPONSE_TOPICS: &str = "tedge/+/commands/res/firmware_upd
 pub struct FirmwareManagerConfig {
     pub tedge_device_id: String,
     pub local_http_host: String,
-    pub tmp_dir: PathBuf,
-    pub data_dir: PathBuf,
-    pub cache_dir: PathBuf,
-    pub file_transfer_dir: PathBuf,
-    pub firmware_dir: PathBuf,
+    pub tmp_dir: Utf8PathBuf,
+    pub data_dir: DataDir,
+    pub cache_dir: Utf8PathBuf,
+    pub file_transfer_dir: Utf8PathBuf,
+    pub firmware_dir: Utf8PathBuf,
```

Comment:
This should be only a `DataDir` as all these paths are derived from it.

Reply:
Done